Fluid/MemberAccessStrategyExtensions.cs

```diff
@@ -162,6 +163,51 @@ public static void Register<T>(this IMemberAccessStrategy strategy, Func<T, stri
         }
 
         /// <summary>
+        /// Registers a type with a <see cref="Func{T, string, Task{Object}}"/> to retrieve any of
+        /// its property values.
+        /// </summary>
+        /// <param name="type">The type to register.</param>
+        /// <param name="accessor">The <see cref="Func{T, string, Task{Object}}"/> instance used to retrieve the value.</param>
+        public static void Register<T>(this IMemberAccessStrategy strategy, Func<T, string, Task<object>> accessor)
+        {
+            Register<T>(strategy, (obj, name, ctx) => accessor(obj, name));
+        }
+
+        /// <summary>
+        /// Registers a type with a <see cref="Func{T, string, TemplateContext, Task{Object}}"/> to retrieve any of
+        /// its property values.
+        /// </summary>
+        /// <param name="type">The type to register.</param>
+        /// <param name="accessor">The <see cref="Func{T, string, TemplateContext, Task{Object}}"/> instance used to retrieve the value.</param>
+        public static void Register<T>(this IMemberAccessStrategy strategy, Func<T, string, TemplateContext, Task<object>> accessor)
+        {
+            strategy.Register(typeof(T), "*", new AsyncDelegateAccessor<T>(accessor));
+        }
+
+        /// <summary>
+        /// Registers a type with a <see cref="Func{T, string, Task{Object}}"/> to retrieve any of
+        /// its property values.
+        /// </summary>
+        /// <param name="type">The type to register.</param>
+        /// <param name="name">The name of the property</param>
+        /// <param name="accessor">The <see cref="Func{T, string, Task{Object}}"/> instance used to retrieve the value.</param>
+        public static void Register<T>(this IMemberAccessStrategy strategy, string name, Func<T, string, Task<object>> accessor)
```

Comment:
Accessor should be `Func<T, Task<object>>` (i.e. no `string` arg). 

This argument is redundant because you're explicitly registering this accessor against a given property name, and so you already know the name of the property- no need for it to be passed in again.

**Also**- there's a copy/paste error in the summary of this method- `retrieve any of its property values` should read `retrieve the given property's value` (or similar).

See the methods I added yesterday.